bokehjs/src/lib/models/widgets/autocomplete_input.ts

```diff
@@ -139,9 +139,18 @@ export class AutocompleteInputView extends TextInputView {
         }
 
         const completions: string[] = []
+        const case_sensitive: boolean = this.model.case_sensitive
+        let actest: (t: string, v: string) => boolean;
+        if (case_sensitive) {
+          actest = (t: string, v: string): boolean => { return t.startsWith(v); }
+        } else {
+          actest = (t: string, v: string): boolean => { return t.toLowerCase().startsWith(v.toLowerCase()); }
+        }
+
         for (const text of this.model.completions) {
-          if (text.startsWith(value))
+          if (actest(text, value)) {
```

Comment:
A simpler approach would be to add a normalization function instead, which would be no-op for case sensitive case and `(text) => text.toLowerCase()` otherwise.

Reply:
Good idea, lmk if the new commit works. Thanks again!